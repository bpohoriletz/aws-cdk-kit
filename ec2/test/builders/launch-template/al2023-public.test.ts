import Al2023PublicBuilder from '../../../launch-template-builders/al2023-public-builder';

describe('new Al2023PublicBuilder()', () => {
  let instance: Al2023PublicBuilder;

  beforeEach(() => {
    instance = new Al2023PublicBuilder();
  });

  it('matches default', () => {
    instance.setPublic!().setUserData!().setMachineImage!();

    expect(instance.getResult()).toMatchSnapshot();
  });

  describe('#setName() ', () => {
    it('customizes name of LaunchTemplate', () => {
      instance.setName('LaunchTempalte');

      expect(instance.getResult()).toMatchSnapshot();
    });
  });
});
