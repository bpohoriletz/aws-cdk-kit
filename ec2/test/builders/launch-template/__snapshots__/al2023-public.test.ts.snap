// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new Al2023PublicBuilder() #setName()  customizes name of LaunchTemplate 1`] = `
LaunchTemplateProduct {
  "launchTemplateName": "LaunchTempalte",
}
`;

exports[`new Al2023PublicBuilder() matches default 1`] = `
LaunchTemplateProduct {
  "associatePublicIpAddress": true,
  "machineImage": AmazonLinux2023ImageSsmParameter {
    "cachedInContext": false,
    "props": {
      "cachedInContext": false,
      "parameterName": "/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-6.1-x86_64",
      "userData": undefined,
    },
  },
  "userData": LinuxUserData {
    "lines": [
      "echo Hello World",
    ],
    "onExitLines": [],
    "props": {},
  },
}
`;
